packages/coordinator/src/screens/manageAppointmentsScreen/ManageAppointmentsScreenContainer.tsx

```diff
@@ -39,6 +40,19 @@ export default function ManageAppointmentsScreenContainer() {
     return CoordinatorFunctions.deleteAppointment(appointmentId);
   };
 
+  const onRemoveDonor = (appointmentId: string) => {
+    setAppointmentsResponse({
+      ...appointmentsResponse,
+      appointments: appointmentsResponse.appointments.map((appointment) => {
+        if (appointment.id !== appointmentId) {
+          return appointment;
+        }
+        return AppointmentUtils.removeDonorFromAppointment(appointment);
```

Comment:
Is the process removing the donor twice,
once for the frontend and one via Firebase?

Why not amend the appointments and load them again- keeping a single source of truth?

Reply:
Well, we could, but that means higher latency. Why is it butter, when both BE and FE use the same util?